doc/model_persistence.rst

```diff
@@ -1,294 +1,307 @@
+<skops_persistence>` which gives you back the python object, but unlike
+`pickle` based persistence solutions, loading the persisted model doesn't
+automatically allow arbitrary code execution. Note that this requires manual
+investigation of the persisted file, which :mod:`skops.io` allows you to do.
+
+The other solutions assume you absolutely trust the source of the file to be
+loaded, as they are all susceptible to arbitrary code execution upon loading
+the persisted file since they all use the pickle protocol under the hood.
+
+3. Do you care about the performance of loading the model, and sharing it
+   between processes where a memory mapped object on disk is beneficial?
+
+If yes, then you can consider using :ref:`joblib <pickle_persistence>`. If this
+is not a major concern for you, then you can use the built-in :mod:`pickle`
+module.
+
+4. Did you try :mod:`pickle` or :mod:`joblib` and found that the model cannot
+   be persisted? This can happen for instance when you have user defined
+   functions in your model.
+
+If yes, then you can use `cloudpickle`_ which can serialize certain objects
+which cannot be serialized by :mod:`pickle` or :mod:`joblib`.
+
+
+Workflow Overview
+-----------------
+
+In a typical workflow, the first step is to train the model using scikit-learn
+and scikit-learn compatible libraries. Note that support for scikit-learn and
+third party estimators varies across the different persistence methods.
+
+Train and Persist the Model
+...........................
+
+Creating an appropriate model depends on your use-case. As an example, here we
+train a :class:`sklearn.ensemble.HistGradientBoostingClassifier` on the iris
+dataset::
+
+  >>> from sklearn import ensemble
+  >>> from sklearn import datasets
+  >>> clf = ensemble.HistGradientBoostingClassifier()
+  >>> X, y = datasets.load_iris(return_X_y=True)
+  >>> clf.fit(X, y)
+  HistGradientBoostingClassifier()
+
+Once the model is trained, you can persist it using your desired method, and
+then you can load the model in a separate environment and get predictions from
+it given input data. Here there are two major paths depending on how you
+persist and plan to serve the model:
+
+- :ref:`ONNX <onnx_persistence>`: You need an `ONNX` runtime and an environment
+  with appropriate dependencies installed to load the model and use the runtime
+  to get predictions. This environment can be minimal and does not necessarily
+  even require `python` to be installed.
+
+- :mod:`skops.io`, :mod:`pickle`, :mod:`joblib`, `cloudpickle`_: You need a
+  python environment with the appropriate dependencies installed to load the
+  model and get predictions from it. This environment should have the same
+  packages and the same versions as the environment where the model was
+  trained. Note that none of these methods support loading a model trained with
+  a different version of scikit-learn, and possibly different versions of other
```

Comment:
However, we should be able to load on a different hardware. This could be potentially meaningful to state for people training on a powerful computer and serving on a much less powerful device (I recall @lesteve fixing the pickle for this use-case).